lib/Sema/TypeCheckType.cpp

```diff
@@ -2221,8 +2221,21 @@ Type TypeResolver::resolveAttributedType(TypeAttributes &attrs,
 
     for (auto i : FunctionAttrs) {
       if (attrs.has(i)) {
-        TC.diagnose(attrs.getLoc(i), diag::attribute_requires_function_type,
-                    TypeAttributes::getAttrName(i));
+          if (attrs.has(TAK_escaping)) {
+              auto &SM = TC.Context.SourceMgr;
+              auto loc = attrs.getLoc(TAK_escaping);
```

Comment:
This entire file is using `TypeAttributes` instead of `DeclAttributes` so the ugly way around is the only way around.  It's ripe for refactoring.